src/term/transform/encode_lists.rs

```diff
@@ -92,8 +93,8 @@ impl Pattern {
 
         true
       }
-      Pattern::Ctr(_, pats) => {
-        let mut uses = false;
+      Pattern::Ctr(Name(nam), pats) => {
+        let mut uses = nam == LCONS;
```

Comment:
nam == LCONS || nam == LNIL

If we write the pattern like this `(Head (LNil)) = LNil` then it will be understood as a constructor by the parser without needing the delayed name resolving